safehttp/flightvalues_test.go

```diff
@@ -0,0 +1,103 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package safehttp_test
+
+import (
+	"context"
+	"fmt"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-safeweb/safehttp"
+	"github.com/google/safehtml"
+)
+
+type safeHeadersInterceptor struct{}
+
+func (ip *safeHeadersInterceptor) Before(w safehttp.ResponseWriter, r *safehttp.IncomingRequest, cfg safehttp.InterceptorConfig) safehttp.Result {
+	// We claim the header here in order to protect it from being tampered. The
+	// only way to set it is through a helper method exposed by this package. It
+	// only allows for setting safe values.
+	h := w.Header().Claim("Super-Safe-Header")
+	safehttp.FlightValues(r.Context()).Put(safeHeaderKey{}, h)
+	return safehttp.NotWritten()
+}
+
+func (ip *safeHeadersInterceptor) Commit(w safehttp.ResponseHeadersWriter, r *safehttp.IncomingRequest, resp safehttp.Response, cfg safehttp.InterceptorConfig) {
+}
+
+func (ip *safeHeadersInterceptor) Match(_ safehttp.InterceptorConfig) bool {
+	// This interceptor does not offer any configuration options.
+	return false
+}
+
+type safeHeaderKey struct{}
+
+func SetHeaderSafely(ctx context.Context, level int) {
+	var value string
+	switch level {
+	case 0:
+		value = "Safe"
+	case 1:
+		value = "VerySafe"
+	case 2:
+		value = "VeryVerySafe"
+	default:
+		value = "Safe"
+	}
+	safehttp.FlightValues(ctx).Get(safeHeaderKey{}).(func([]string))([]string{value})
```

Comment:
This can be broken apart to be more descriptive/readable:

setter := safehttp.FlightValues(ctx).Get(safeHeaderKey{}).(func([]string))
setter([]string{value})

Reply:
Done in 0ce62ea.